DirectProgramming/C++SYCL_FPGA/ReferenceDesigns/svd/README.md

```diff
@@ -0,0 +1,354 @@
+
+![](assets/SVD.svg)
+
+### Source file structure
+| File name              | Type                            | Content                                                                                                                                                                                              |
+| ---------------------- | ------------------------------- | ---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| svd_demo.cpp           | Host code                       | Launch a demonstration using the SVD design. Contains the `main()` function.                                                                                                                         |
+| svd_testcase.hpp       | Host code                       | A `struct` that lunches the design with a set of input and check result for correctness. The input can either be specified or generated.                                                             |
+| svd_testbench_tool.hpp  | Host code                       | Helper functions that is used in the test bench of the SVD demonstration to check for correctness.                                                                                                   |
+| print_matrix.hpp       | Host code                       | Helper functions to print matrices.                                                                                                                                                                  |
+| golden_pca.hpp         | Host code                       | A CPU reference PCA design that is used to calculate reference eigen values for the test bench.                                                                                                      |
+| svd.hpp                | Host code with some device code | Contains wrapper function `SingularValueDecomposition` that launches individual kernels of the SVD design.                                                                                           |
+| memory_transfers.hpp   | Device code                     | Contains kernel implementations of `MatrixReadFromDDRTo2PipesByBlocks` , `MatrixReadPipeToDDR` and `VectorReadPipeToDDR`. These kernels transfers data between DDR and streaming interfaces (pipes).|
+| post_process.hpp       | Device code                     | Contains kernel implementation of `USVFromEigens`.                                                                                                                                                   |
+### Input covariance matrix computation
+The covariance computation in this design is the same as used in the [PCA](../pca/README.md) reference design except without standardization.
```

Comment:
mention that `fpga_linalg::StreamingCovarianceMatrix` is re-used to obtain this computation